mono/metadata/class.c

```diff
@@ -1570,7 +1570,11 @@ mono_class_setup_fields (MonoClass *klass)
 	if (gtd) {
 		mono_class_setup_fields (gtd);
 		if (mono_class_has_failure (gtd)) {
-			mono_class_set_type_load_failure (klass, "");
+			MonoError gtd_error;
```

Comment:
It’d be good to factor this pattern into a function, as it’s repeated several times.